onsp_co2_employee_commuting/models/co2_commuting.py

```diff
@@ -0,0 +1,18 @@
+
+from odoo import models, fields, api
+
+
+
+class Co2Commuting(models.Model):
+    _name = "co2.commuting"
+
+
+    carbon_factor_id = fields.Many2one('carbon.factor', domain="[('id', 'in', allowed_carbon_factor_ids),('carbon_value', '!=', False)]", string="Transport mean")
+    allowed_carbon_factor_ids = fields.Many2many('carbon.factor', related="hr_employee_id.company_id.employee_commuting_carbon_factor.descendant_ids", domain="[('carbon_uom_id', 'ilike', 'km')]")
```

Comment:
I wouldn't use a "ilike" operator for the UOM, could lead to unexpected results even if the probability is very low.

```suggestion
    allowed_carbon_factor_ids = fields.Many2many('carbon.factor', related="hr_employee_id.company_id.employee_commuting_carbon_factor.descendant_ids", domain=lambda self: [('carbon_uom_id', '=', self.env.ref('uom.product_uom_km').id)])
```

[See an example here](https://github.com/odoo/odoo/blob/52fe645be92bf3c30edbab39bc1dc5a1ac4f1bd0/addons/hr_timesheet/models/res_company.py#L37)